### DISCUSSION TITLE
Terms for build vs runtime configuration in Quarkus dist

### DISCUSSION THREAD

**Discussion by stianst** [keycloak/keycloak#9709] [Archive]
We don't currently have strict/common terms to refer to build vs runtime configuration, and need to define some common terminology here as it would not be ideal if one guide refers to "options limited to build-time" while another guide calls it "build option".

The proposal here is:

* *Build options*: everything that affects the build
* *Configuration*: everything that can be configured when running Keycloak

Most people will expect that _configuration_ can be done at runtime, without doing a __build__ first, so using this terminology uses the standard term for __configuration__, while a separate term for __special__ configuration that requires a build.

An example of using these terms in a sentence:

__To configure the database first you need to select the vendor with the --db build option. Then you need to configure the URL with the db-url configuration option.__

An example of using these terms on the options search page:

![image](https://user-images.githubusercontent.com/2271511/150510376-1f58ec65-8e33-45c3-ab82-6ed8a98c344d.png)

**Comment by andreaTP** [comment]
I like this proposal and the usability of the mocked search page.

Just as a reference, in the Quarkus project, they are marking "build time options" with a lock 🔒 :
https://quarkus.io/guides/config#build-time-configuration

**Reply by stianst** [reply]
Search page is real, but I changed the label names for the above screenshot though:
https://github.com/keycloak/keycloak-web/pull/253

We're also already marking build options with the lock, that's already merged.

**Reply by stianst** [reply]
Actually, changing the icon to a tool icon, as I think that represents "build" better than a lock:
https://github.com/keycloak/keycloak/pull/9717

**Comment by sschu** [comment]
I also like the proposal. One could maybe add then when referring to a specific parameter, the terms `build option` and `configuration option` should be used. But maybe that just comes naturally and doesn't need to be specified explicitly.
